modules/internal/ChapelError.chpl

```diff
@@ -177,12 +187,54 @@ module ChapelError {
   }
 
   pragma "no doc"
+  proc chpl_error_type_name(err: Error) : string {
+    var cid =  __primitive("getcid", err);
+    var nameC: c_string = __primitive("class name by id", cid);
+    var nameS = nameC:string;
+    return nameS;
+  }
+  pragma "no doc"
+  proc chpl_describe_error(err: Error) : string {
+    var nameS = chpl_error_type_name(err);
+
+    var ret = nameS + ": " + err.message();
+
+    return ret;
+  }
+
+  pragma "no doc"
+  pragma "insert line file info"
+  proc chpl_save_line_in_error(err: Error) {
+    const line = __primitive("_get_user_line");
+    const fileId = __primitive("_get_user_file");
+    err.thrownLine = line;
+    err.thrownFileId = fileId;
+  }
+  pragma "no doc"
   proc chpl_delete_error(err: Error) {
     delete err;
   }
+  pragma "no doc"
   pragma "function terminates program"
+  pragma "insert line file info"
   proc chpl_uncaught_error(err: Error) {
-    var tmpstring = "uncaught error - " + stringify(err);
-    __primitive("chpl_error", tmpstring.c_str());
+    extern proc chpl_error_preformatted(c_string);
+
+    const myFileC:c_string = __primitive("chpl_lookupFilename",
+                                         __primitive("_get_user_file"));
+    const myFileS = myFileC:string;
+    const myLine = __primitive("_get_user_line");
```

Comment:
I might call these `uncaught` instead of `my` for clarity, and then swap them with the `thrown` section since that's the order they're printed in. It's fine as it is though.